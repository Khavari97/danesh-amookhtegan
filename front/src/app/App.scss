@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


