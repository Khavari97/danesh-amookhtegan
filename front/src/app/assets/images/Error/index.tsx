import errorImage from './file/ErrorImage.svg';

const error = {
    errorImage
}

export default error