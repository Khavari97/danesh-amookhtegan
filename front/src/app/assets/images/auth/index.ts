import clock from './files/clock.svg'
import darkGreenDown from './files/dark-green-down.svg'
import darkGreenTop from './files/dark-green-top.svg'
import google from './files/google.svg'
import greenBorderDown from './files/green-border-down.svg'
import lightGreenDown from './files/light-green-down.svg'
import lightGreenTop from './files/light-green-top.svg'
import receivedEmail from './files/received-email.svg'
import signInImage from './files/signin-image.svg'
import signUpImage from './files/signup-image.svg'
import backgroundGroup from './files/background-group.svg'
import home from './files/home.svg'
import search from './files/search.svg'


const auth = {
    clock,
    darkGreenDown,
    darkGreenTop,
    google,
    greenBorderDown,
    lightGreenDown,
    lightGreenTop,
    receivedEmail,
    signInImage,
    signUpImage,
    backgroundGroup,
    home,
    search
}

export default auth