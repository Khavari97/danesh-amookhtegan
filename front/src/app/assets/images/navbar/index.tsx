import search from './files/search.svg'
import bell from './files/bell.svg'
import front_desk from './files/front_desk.svg'
import home from './files/home.svg'
import mainpage from './files/mainpage.svg'
import registration from './files/registration.svg'

const navbar = {
    search,
    bell,
    front_desk,
    home,
    mainpage,
    registration
}
export default navbar
