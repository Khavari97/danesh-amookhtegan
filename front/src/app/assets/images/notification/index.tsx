import errorIcon from './files/error-icon.svg';
import closeImage from './files/close-image.svg';

const notification = {
    errorIcon,
    closeImage
}

export default notification