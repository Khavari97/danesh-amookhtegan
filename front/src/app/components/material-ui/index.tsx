import button from "./button/button"
import icon from "./icon/icon"
import input from "./input/input"
import grid from "./grid/grid"

export const Button = (props: any) => button(props)
export const Icon = (props: any) => icon(props)
export const Input = (props: any) => input(props)
export const Grid = (props: any) => grid(props)
