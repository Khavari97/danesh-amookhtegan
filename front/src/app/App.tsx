import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
