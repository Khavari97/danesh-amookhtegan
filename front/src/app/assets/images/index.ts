import auth from "./auth";
import notification from "./notification";
import error from "./Error";
import userProfile from "./userProfile"
import navbar from "./navbar"
const images = {
    auth,
    notification,
    error,
    userProfile,
    navbar
}

export default images