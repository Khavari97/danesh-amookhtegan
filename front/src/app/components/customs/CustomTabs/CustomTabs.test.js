import React from "react"
import { shallow, configure, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import App, { TabPanel } from "./CustomTabs"
import { Icon } from "../../material-ui";
import images from "../../../assets/images/userProfile/index"
import renderer from "react-test-renderer";
import { AppBar, Box, Tab, Tabs, withStyles } from '@material-ui/core';

const skillIcon = (
  <Icon>
    <img style={{ width: '30px' }} alt="development_skill" src={images.development_skill} data-test='development_skill' />
  </Icon>
);
const infoIcon = (
  <Icon>
    <img style={{ width: '30px' }} alt="customer" src={images.customer} data-test='customer' />
  </Icon>
);

const resumeIcon = (
  <Icon>
    <img style={{ width: '30px' }} alt="pdfFile" src={images.pdfFile} data-test='pdfFile' />
  </Icon>
);
configure({ adapter: new Adapter() })


const Div = () => {
  return <div className="containerOne" ></div>
}
const DivTwo = () => {
  return <div className="containerTwo" ></div>
}
const DivThree = () => {
  return <div className="containerThree" ></div>
}

configure({ adapter: new Adapter() })
const DeskTabs = [
  { label: "تکمیل پروفایل", icon: infoIcon, component: <Div /> },
  { label: "مهارت ها", icon: skillIcon, component: <DivTwo /> },
  { label: "رزومه", icon: resumeIcon, component: <DivThree /> },
]

describe('Custom tab component', () => {

  let wrapper = mount(<App />);
  wrapper.setProps({ tabs: DeskTabs });

  it('check props', () => {
    expect(
      wrapper
        .props()
    ).toEqual({ tabs: DeskTabs });
  })

  it('check tabs buttons div', () => {
    expect(
      wrapper
        .find('[aria-label="tabs"]').exists()
    ).toBe(true)
  })

  it('check tabs count buttons ', () => {
    expect(
      wrapper
        .find('[aria-label="tabs"]').find('button').length
    ).toBe(3)
  })
  //
  it('Tab exists', () => {
    expect(
      wrapper
        .find(Tab)
        .exists()
    ).toBe(true);
  })

  it('TabPanel length', () => {
    expect(
      wrapper.find(TabPanel)
        .length
    ).toBe(3);
  })

  //default tab active
  it('check value ', () => {
    expect(
      wrapper.find(TabPanel).at(0).props().value
    ).toBe(0)
  })
  it('snapshot', () => {
    const shot = <App />
    const snapshot = renderer.create(shot).toJSON();
    expect(snapshot).toMatchSnapshot();
  })
})

describe('Simulate tab component Two', () => {

  let wrapper = mount(<App />);
  wrapper.setProps({ tabs: DeskTabs });

  it('check props', () => {
    expect(
      wrapper
        .props()
    ).toEqual({ tabs: DeskTabs });
  })
  // click simulate second tab

  wrapper.find("button#simple-tab-1").simulate('click')

  it('find content div that active-two', () => {
    expect(
      wrapper.find("div.containerTwo")
        .exists()
    ).toEqual(true)
  })

  it('check value ', () => {
    expect(
      wrapper.find(TabPanel).at(0).props().value
    ).toBe(1)
  })

})

describe('Simulate tab component Three', () => {

  let wrapper = mount(<App />);
  wrapper.setProps({ tabs: DeskTabs });

  it('check props', () => {
    expect(
      wrapper
        .props()
    ).toEqual({ tabs: DeskTabs });
  })


  // click simulate second tab

  wrapper.find("button#simple-tab-2").simulate('click')

  it('find content div that active-three', () => {
    expect(
      wrapper.find("div.containerThree")
        .exists()
    ).toEqual(true)
  })

  it('check value ', () => {
    expect(
      wrapper.find(TabPanel).at(0).props().value
    ).toBe(2)
  })

})



