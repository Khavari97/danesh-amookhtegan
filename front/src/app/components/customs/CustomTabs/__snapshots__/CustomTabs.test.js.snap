// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom tab component snapshot 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root WithStyles(ForwardRef(AppBar))-root-1 MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-2"
    >
      <div
        className="MuiTabs-scrollable"
        style={
          Object {
            "height": 99,
            "overflow": "scroll",
            "position": "absolute",
            "top": -9999,
            "width": 99,
          }
        }
      />
      <div
        className="MuiTabs-scroller MuiTabs-scrollable"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="tabs"
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        />
      </div>
    </div>
  </header>
</div>
`;
